server/api.go

```diff
@@ -37,11 +37,15 @@ func (p *Plugin) ServeHTTP(_ *plugin.Context, w http.ResponseWriter, r *http.Req
 
 	router := mux.NewRouter()
 
-	router.HandleFunc("/api/v1/legalhold/list", p.listLegalHolds)
-	router.HandleFunc("/api/v1/legalhold/create", p.createLegalHold)
-	router.HandleFunc("/api/v1/legalhold/{legalhold_id:[A-Za-z0-9]+}/release", p.releaseLegalHold)
-	router.HandleFunc("/api/v1/legalhold/{legalhold_id:[A-Za-z0-9]+}/update", p.updateLegalHold)
-	router.HandleFunc("/api/v1/legalhold/{legalhold_id:[A-Za-z0-9]+}/download", p.downloadLegalHold)
+	// Routes called by the plugin's webapp
+	router.HandleFunc("/api/v1/legalhold/list", p.listLegalHolds).Methods(http.MethodGet)
+	router.HandleFunc("/api/v1/legalhold/create", p.createLegalHold).Methods(http.MethodPost)
+	router.HandleFunc("/api/v1/legalhold/{legalhold_id:[A-Za-z0-9]+}/release", p.releaseLegalHold).Methods(http.MethodPost)
+	router.HandleFunc("/api/v1/legalhold/{legalhold_id:[A-Za-z0-9]+}/update", p.updateLegalHold).Methods(http.MethodPost)
+	router.HandleFunc("/api/v1/legalhold/{legalhold_id:[A-Za-z0-9]+}/download", p.downloadLegalHold).Methods(http.MethodGet)
+
+	// Debug routes
+	router.HandleFunc("/api/v1/legalhold/debug/run", p.runJobFromAPI).Methods(http.MethodPost)
```

Comment:
Let's make this a legit way to trigger the job.  Not a debug tool.

Reply:
@wiggin77 I agree that `debug` is not the right term here. I was under the impression we wanted this to be only used for demos, and wanted to separate the endpoint from the existing endpoints called by the frontend.

Reply:
@wiggin77 Maybe we should guard this by an env var if it's only expected to be run for demos? If we see value in customers being able to run it via API call that, then no need to do add that.

Reply:
It may be used outside of demos. Also we don't want to create added complexity creating demo environments.